checkov/cloudformation/image_referencer/provider/aws.py

```diff
@@ -0,0 +1,98 @@
+
+    image_repo = find_in_dict(input_dict=resource, key_path="SourceConfiguration/ImageRepository")
+    if isinstance(image_repo, dict):
+        repo_type = image_repo.get("ImageRepositoryType")
+        name = image_repo.get("ImageIdentifier")
+        if name and isinstance(name, str) and repo_type == "ECR_PUBLIC":
+            image_names.append(name)
+
+    return image_names
+
+
+def extract_images_from_aws_batch_job_definition(resource: dict[str, Any]) -> list[str]:
+    image_names: list[str] = []
+
+    properties = extract_json(resource.get("ContainerProperties"))
+    if isinstance(properties, dict):
+        name = properties.get("Image")
+        if name and isinstance(name, str):
+            image_names.append(name)
+
+    node_range = find_in_dict(input_dict=resource, key_path="NodeProperties/NodeRangeProperties")
+    if isinstance(node_range, list):
+        for node in node_range:
+            name = find_in_dict(input_dict=node, key_path="Container/Image")
+            if name and isinstance(name, str):
+                image_names.append(name)
+
+    return image_names
+
+
+def extract_images_from_aws_codebuild_project(resource: dict[str, Any]) -> list[str]:
+    image_names: list[str] = []
+
+    name = find_in_dict(input_dict=resource, key_path="Environment/Image")
+    if name and isinstance(name, str):
+        # AWS provided images have an internal identifier
+        if not name.startswith("aws/codebuild/"):
+            image_names.append(name)
+
+    return image_names
+
+
+def extract_images_from_aws_ecs_task_definition(resource: dict[str, Any]) -> list[str]:
+    image_names: list[str] = []
+
+    definitions = extract_json(resource.get("ContainerDefinitions"))
+    if isinstance(definitions, list):
+        for definition in definitions:
+            name = definition.get("Image")
+            if name and isinstance(name, str):
+                image_names.append(name)
+
+    return image_names
+
+
+def extract_images_from_aws_lightsail_container_service_deployment_version(resource: dict[str, Any]) -> list[str]:
+    image_names: list[str] = []
+
+    containers = find_in_dict(input_dict=resource, key_path="ContainerServiceDeployment/Containers")
+    if isinstance(containers, list):
+        for container in containers:
+            name = container.get("Image")
+            if name and isinstance(name, str):
+                image_names.append(name)
+
+    return image_names
+
+
+# needs to be at the bottom to add the defined functions
+SUPPORTED_AWS_IMAGE_RESOURCE_TYPES = {
+    "AWS::AppRunner::Service": extract_images_from_aws_apprunner_service,
+    "AWS::Batch::JobDefinition": extract_images_from_aws_batch_job_definition,
+    "AWS::CodeBuild::Project": extract_images_from_aws_codebuild_project,
+    "AWS::ECS::TaskDefinition": extract_images_from_aws_ecs_task_definition,
+    "AWS::Lightsail::Container": extract_images_from_aws_lightsail_container_service_deployment_version,
```

Comment:
what about:
EKS, Lambda, ECR, Copilot?

Reply:
EKS: you can only create cluster, node groups and some other management stuff, nothing related to containers. those are added via other means like normal Kubernetes manifests or helm
Lambda: only allows privately stored images in ECR, which can't scan anyway
ECR: only defines the registry and its settings, but you can' reference any images in the resources
Copilot: is a CLI tool it has its own configuration files